pkg/services/authn/clients/provisioning.go

```diff
@@ -0,0 +1,62 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	claims "github.com/grafana/authlib/types"
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/pkg/services/authn"
+)
+
+var (
+	_ authn.ContextAwareClient = (*Provisioning)(nil)
+	_ fmt.Stringer             = (*Provisioning)(nil) // for debugging
+)
+
+type Provisioning struct {
+	webhookRegexp *regexp.Regexp
+}
+
+func ProvideProvisioning() *Provisioning {
+	// It's fine to compile a regexp here. The function is only called once per instance of APIBuilder, of which there should only ever be 1.
+	// Assumption: APIVERSION has no leading or trailing slashes.
+	webhookRegexp := regexp.MustCompile("^/apis/" + regexp.QuoteMeta(provisioning.APIVERSION) + "/namespaces/[^/]+/repositories/[^/]+/webhook$")
+	return &Provisioning{webhookRegexp}
+}
+
+func (p *Provisioning) String() string {
+	return p.Name()
+}
+
+func (*Provisioning) Name() string {
+	return authn.ClientProvisioning
+}
+
+func (p *Provisioning) Authenticate(ctx context.Context, r *authn.Request) (*authn.Identity, error) {
+	return &authn.Identity{
```

Comment:
do we need to always replace the user?  if a user exists in context (not anon) should we use it?